src/ir/types/type.h

```diff
@@ -31,13 +46,13 @@ class Type {
   const TypeBase& type_base() const { return *type_.get(); }
```

Comment:
Is this PR the right time to add `operator*` and `operator->` instead of using `type_base`?

Reply:
Let us do it in a separate PR. I want to get rid of `type_base` if possible and `Type` as the type-erased (no pun intended :) ) version of `TypeBase`.

Reply:
Sounds good :+1: